packages/graphql-typescript-definitions/src/index.ts

```diff
@@ -92,6 +93,8 @@ export class Builder extends EventEmitter {
   >();
 
   private filesystem: GraphQLFilesystem | undefined;
+  private readonly schemaCache = new Map<string, Promise<GraphQLSchema>>();
```

Comment:
Our largest schema take 3.5 - 8 seconds to read (depending on CPU load). So caching occupies a little memory, but saves at least 3.5 seconds during `handleSchemaUpdate`. It also cuts `handleDocumentUpdate`'s time in half.